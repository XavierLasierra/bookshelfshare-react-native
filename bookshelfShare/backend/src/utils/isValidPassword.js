const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
