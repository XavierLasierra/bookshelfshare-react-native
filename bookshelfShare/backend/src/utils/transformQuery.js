function transformQuery(query) {
  let transformedQuery = {};
  const entries = Object.entries(query);
  if (entries.length > 0) {
    entries.forEach(([property, filter]) => {
      if (property.includes('_')) {
        const [propertyName, symbol] = property.split('_');
        transformedQuery = { ...transformedQuery, [propertyName]: { [`$${symbol}`]: filter } };
      } else {
        transformedQuery = { ...transformedQuery, [property]: { $regex: filter, $options: 'i' } };
      }
    });
  }
  return transformedQuery;
}

module.exports = { transformQuery };
