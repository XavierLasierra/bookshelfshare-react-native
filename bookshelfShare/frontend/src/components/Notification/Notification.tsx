import React, {useState, useEffect} from 'react';
import {Snackbar} from 'react-native-paper';
import {useSelector} from 'react-redux';
import {IStore} from '../../types/interfaces';

export default function Notification() {
  const [visible, setVisible] = useState(false);
  const notificationMessage = useSelector(
    (store: IStore) => store.notifications,
  );

  useEffect(() => {
    if (notificationMessage[0]) {
      setVisible(true);
    }
  }, [notificationMessage]);

  function onDismissSnackBar() {
    setVisible(false);
  }

  return (
    <Snackbar
      visible={visible}
      onDismiss={onDismissSnackBar}
      action={{
        testID: 'closeSnackbarButton',
        label: 'x',
        onPress: onDismissSnackBar,
      }}>
      {[notificationMessage]}
    </Snackbar>
  );
}
