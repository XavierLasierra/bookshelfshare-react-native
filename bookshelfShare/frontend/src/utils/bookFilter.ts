export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
