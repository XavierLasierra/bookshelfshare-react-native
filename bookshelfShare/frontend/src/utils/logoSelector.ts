import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
