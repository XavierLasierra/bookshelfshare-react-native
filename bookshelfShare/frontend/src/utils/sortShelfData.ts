export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
