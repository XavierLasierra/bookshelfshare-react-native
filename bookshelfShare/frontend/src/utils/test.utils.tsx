import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
