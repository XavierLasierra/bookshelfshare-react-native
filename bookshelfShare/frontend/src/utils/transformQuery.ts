interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
