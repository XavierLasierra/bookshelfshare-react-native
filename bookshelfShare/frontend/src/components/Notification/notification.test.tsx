import React from 'react';
import Notification from './Notification';
import { render, fireEvent } from '../../utils/test.utils';

jest.mock('react-native', () => {
  const reactNativeMock = jest.requireActual('react-native');

  reactNativeMock.Animated.timing = (value: any, config: any) => ({
    start: (callback: any) => {
      value.setValue(config.toValue);
      callback({ finished: true });
    }
  });

  return reactNativeMock;
});

describe('Given a Notification component', () => {
  describe('When it is rendered', () => {
    describe('And there is not a notification in store', () => {
      test('Then should match the snapshot', () => {
        const screen = render(<Notification />);
        expect(screen).toMatchSnapshot();
      });
    });
    describe('And there is a notification in store', () => {
      let screen: any;
      beforeEach(() => {
        const initialState = { notifications: ['Hi'] };
        screen = render(<Notification />, initialState);
      });
      test('Then should match the snapshot', () => {
        expect(screen).toMatchSnapshot();
      });

      describe('And you click on closeSnackbarButton button', () => {
        test('Then should match the snapshot', () => {
          const closeSnackbarButton = screen.getByTestId('closeSnackbarButton');
          fireEvent.press(closeSnackbarButton);
          expect(screen).toMatchSnapshot();
        });
      });
    });
  });
});
