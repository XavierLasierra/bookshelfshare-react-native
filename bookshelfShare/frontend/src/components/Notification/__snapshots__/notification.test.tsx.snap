// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Notification component When it is rendered And there is a notification in store And you click on closeSnackbarButton button Then should match the snapshot 1`] = `null`;

exports[`Given a Notification component When it is rendered And there is a notification in store Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLiveRegion="polite"
    collapsable={false}
    nativeID="animatedComponent"
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "borderRadius": 4,
        "elevation": 6,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 8,
        "opacity": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 6,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "flex": 1,
              "flexWrap": "wrap",
              "marginLeft": 16,
              "marginVertical": 14,
            },
            Object {
              "color": "#ffffff",
              "marginRight": 0,
            },
          ],
        ]
      }
    >
      Hi
    </Text>
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 0,
          "marginHorizontal": 8,
          "marginVertical": 6,
          "minWidth": "auto",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
        testID="closeSnackbarButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  Object {
                    "marginHorizontal": 8,
                  },
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#03dac4",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            x
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Given a Notification component When it is rendered And there is not a notification in store Then should match the snapshot 1`] = `null`;
